### DISCUSSION TITLE
Health probes for non-web apps using distroless base images like Ubuntu Chiseled

### DISCUSSION THREAD

**Discussion by OGCdefi** [dotnet/dotnet-docker#5728] [Q&A] [answered]
I have the following scenario: my team has a background worker app, it's a .NET 8 console app, NOT a web app nor service. It does not define end web endpoint. It is running in Kubernetes and I need to implement a liveliness probe. In the past (before we moved to the Ubuntu Chiseled distroless base image), we had a simple probe that would observe for a file. The probe used the "find" command.

The probe will not work anymore, as the distroless image doesn't have "find" nor bash. 

What's the recommended way to implement a probe for this kind of apps? 

I did a search on the web and in discussion section and the only related topic was this: https://github.com/dotnet/dotnet-docker/discussions/4296
But that's a different case, because the app in question is a web api that can simply implement a health check endpoint.

**Comment by mthalman** [comment]
If you implement a custom app that would be used to run your probe logic, can't it replicate what you were doing before without using Bash? That's effectively what the response to https://github.com/dotnet/dotnet-docker/discussions/4296 is suggesting.

**Reply by OGCdefi** [reply]
I guess I didn't clarify this: when I said we had a simple probe I mean in Kubernetes manifests, we had something like this: 

```
livenessProbe:
  exec:
    command:
    - find 
    - /tmp/healthy
    - -mmin 
    - '-1'
```
The app will simple write to the /tmp/healthy file every X number of seconds (this doesn't rely on bash). The problem arises when defined the probe in the Kubernetes manifests, because I don't have a command to invoke to check for the file anymore.

**Reply by mthalman** [reply, marked as answer]
AFAIK, `command` doesn't rely on Bash. So if you had a custom tool installed in your container, as was previously suggested, you could target that tool executable directly with your `command` configuration here.

**Reply by OGCdefi** [reply]
Thanks for the input @mthalman. You got it right. I ended implementing a really simply dotnet console app that would check in the "/tmp/healthy" file and exit with status code 0 if everything was right or 1 otherwise. Since this is a .NET app, it will execute just fine using the "dotnet" command. 
The probe would look like this: 

```yaml
livenessProbe:
  exec:
    command:
    - dotnet
    - "folder/to/custom/probe.dll"
    - /tmp/healthy
```
